internal/controller/controllers/agent_controller.go

```diff
@@ -333,14 +488,29 @@ func (r *AgentReconciler) updateStatus(ctx context.Context, log logrus.FieldLogg
 	} else {
 		setConditionsUnknown(agent)
 	}
+	if agent.Status.DebugInfo.State == models.HostStatusAddedToExistingCluster &&
+		agent.Status.Progress.CurrentStage == models.HostStageRebooting {
+		alreadyApproved, requeue, err := r.tryApproveDay2CSRs(ctx, agent)
```

Comment:
why not to put logic of re queuing inside  tryApproveDay2CSRs ? I mean why to return boolean if you can return relevant object and remove most of the logic from here?

Reply:
I preferred the tryApproveDay2CSRs just to manage the approval.  I didn't want it to handle the requeue and managing stage (rebooting/done) which is the flag if we want to continue the approval cycle. Only updateStatus should touch and modify the agent.

Reply:
In general I think that we should separate responsibilities: updateStatus updates the agent returns the return value to reconcile, and tryApproveDay2CSRs just tries to approve.